src/main/kotlin/de/bund/digitalservice/useid/identification/IdentificationSessionController.kt

```diff
@@ -116,38 +102,20 @@ class IdentificationSessionsController(
         authentication: Authentication,
     ): ResponseEntity<GetResultResponseType> {
         val tenant = authentication.details as Tenant
+        validateTenant(tenant, eIdSessionId)
 
-        val userData: GetResultResponseType?
-        val identificationSession = identificationSessionService.findByEIdSessionId(eIdSessionId)
-            ?: return ResponseEntity.status(HttpStatus.NOT_FOUND).build()
-        if (tenant.id != identificationSession.tenantId) {
-            log.error("Tenant is not the tenant used to start the identification session")
-            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build()
-        }
-        try {
-            val eidService = EidService(eidServiceConfig)
-            userData = eidService.getEidInformation(eIdSessionId.toString())
-            metricsService.incrementSuccessCounter(METRIC_NAME_EID_INFORMATION, tenant.id)
-        } catch (e: Exception) {
-            metricsService.incrementErrorCounter(METRIC_NAME_EID_INFORMATION, tenant.id)
-            log.error("Failed to fetch identity data: ${e.message}")
-            throw e
-        }
-
-        // resultMajor for success can be found in TR 03112 Part 1 -> Section 4.1.2 ResponseType
-        if (userData.result.resultMajor.equals("http://www.bsi.bund.de/ecard/api/1.1/resultmajor#ok")) {
-            try {
-                identificationSessionService.delete(identificationSession)
-            } catch (e: Exception) {
-                log.error("Failed to delete identification session. id=${identificationSession.id}")
-            }
-        } else {
-            // resultMinor error codes can be found in TR 03130 Part 1 -> 3.4.1 Error Codes
-            log.info("The resultMinor for identification session is ${userData.result.resultMinor}. id=${identificationSession.id}")
-        }
+        val identity = identificationSessionService.getIdentity(eIdSessionId, tenant.id)
         return ResponseEntity
             .status(HttpStatus.OK)
             .contentType(MediaType.APPLICATION_JSON)
-            .body(userData)
+            .body(identity)
+    }
+
+    fun validateTenant(tenant: Tenant, eIdSessionId: UUID) {
+        val identificationSession = identificationSessionService.findByEIdSessionId(eIdSessionId)
+            ?: throw IdentificationSessionNotFoundException()
```

Comment:
Is there any place where we would not throw the IdentificationSessionNotFounException after calling findByEIdSessionId? If so, we could maybe encapsulate that part?